pkg/controller/vitessshardreplication/vitessshardreplication_controller.go

```diff
@@ -226,3 +214,22 @@ func (r *ReconcileVitessShard) Reconcile(request reconcile.Request) (reconcile.R
 	reconcileCount.WithLabelValues(metricLabels(vts, err)...).Inc()
 	return result, err
 }
+
+func (r *ReconcileVitessShard) initReplication(ctx context.Context, vts *planetscalev2.VitessShard, wr *wrangler.Wrangler, rb *results.Builder) {
```

Comment:
We don't usually pass resultBuilders around. Make a new one inside this function and return `resultBuilder.Result()`. It's up to the caller if they want to merge that with their own resultBuilder.

Reply:
Done.